src/components/tl/ucp/alltoallv/alltoallv_pairwise.c

```diff
@@ -102,6 +121,19 @@ ucc_status_t ucc_tl_ucp_alltoallv_pairwise_start(ucc_coll_task_t *coll_task)
 
     task->super.super.status = UCC_INPROGRESS;
     task->n_polls            = ucc_min(1, task->n_polls);
+
+    if (UCC_TL_UCP_TEAM_CTX(team)->cfg.pre_reg_mem) {
+        ucc_tl_ucp_pre_register_mem(team, task->args.src.info_v.buffer,
```

Comment:
Is it safe to register the whole range here? In alltoall we know that buffer is contig and it's ok to register it, but in alltoallv there are might be holes somewhere in the middle with unused and unallocated memory, how ucp_mem_map will handle this? Maybe instead we should check all displacements to make sure that all offsets are part of one large allocation or register only base address in alltoallv if it's possible?

Reply:
worth discussion in WG. What we define in spec - do we allow arbitrary offsets from address 0, or all displs should be in the single buffer

Reply:
i already use offsets from 0 in the application, it's very convenient because it allows to save copy in and copy out if data is not contig. Maybe consider add coll arg flag e.g. UCC_COLL_ARGS_FLAG_SRC_CONTIG, UCC_COLL_ARGS_DST_CONTIG?

Reply:
> Is it safe to register the whole range here? In alltoall we know that buffer is contig and it's ok to register it, but in alltoallv there are might be holes somewhere in the middle with unused and unallocated memory, how ucp_mem_map will handle this? Maybe instead we should check all displacements to make sure that all offsets are part of one large allocation or register only base address in alltoallv if it's possible?

if that( each buffer is from a different allocation) the case, in cuda case it will just register the  first buffer region. it won't fail but may not affect.

Reply:
added flags to indicate src/dst contig or not?